src/main/java/com/example/eatmate/app/domain/notice/dto/NoticeResponseDto.java

```diff
@@ -0,0 +1,37 @@
+package com.example.eatmate.app.domain.notice.dto;
+
+import java.time.LocalDateTime;
+
+import com.example.eatmate.app.domain.notice.domain.Notice;
+
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+public class NoticeResponseDto {
+
+	Long noticeId;
+
+	String title;
+
+	String content;
+
+	LocalDateTime createdAt;
+
+	LocalDateTime updatedAt;
+
+	@Builder
+	private NoticeResponseDto(Notice notice) {
+		this.noticeId = notice.getId();
+		this.title = notice.getTitle();
+		this.content = notice.getContent();
+		this.createdAt = notice.getCreatedAt();
+		this.updatedAt = notice.getUpdatedAt();
+	}
+
+	public static NoticeResponseDto createNoticeResponseDto(Notice notice) {
+		return NoticeResponseDto.builder()
+			.notice(notice)
+			.build();
+	}
+}
```

Comment:
현재 코드도 잘 작성되어 있지만, 불변성과 명시성을 좀 더 높일 수 있을 것 같습니다!

1. 필드에 final 키워드 추가로 불변성 확보
2. 정적 팩토리 메서드 이름을 관례적으로 많이 사용하는 'from'으로 변경
3. 빌더 패턴 사용 시 필드를 좀 더 명시적으로 보여주기

현재 코드도 충분히 잘 작동하니 선택적으로 반영해 주시면 될 것 같습니다! 😊

```suggestion
@Getter
public class NoticeResponseDto {
   private final Long noticeId;
   private final String title;
   private final String content;
   private final LocalDateTime createdAt;
   private final LocalDateTime updatedAt;

   @Builder
   private NoticeResponseDto(Long noticeId, String title, String content, 
                           LocalDateTime createdAt, LocalDateTime updatedAt) {
       this.noticeId = noticeId;
       this.title = title;
       this.content = content;
       this.createdAt = createdAt;
       this.updatedAt = updatedAt;
   }

   public static NoticeResponseDto from(Notice notice) {
       return NoticeResponseDto.builder()
           .noticeId(notice.getId())
           .title(notice.getTitle())
           .content(notice.getContent())
           .createdAt(notice.getCreatedAt())
           .updatedAt(notice.getUpdatedAt())
           .build();
   }
}
```

Reply:
혹시 from 메소드는 단일 책임 원칙(SRP)에 위배되지는 않는건가요?

Reply:
그리고 Notice에도 createNotice 메소드가 있는데 이부분도 from으로 바꾸는게 나을까요?

Reply:
from 메소드는 Notice 엔티티를 DTO로 변환하는 단일 책임만을 수행하고 있습니다! 이는 DTO의 주요 책임 중 하나이므로, SRP에 위배되지 않는것 같습니다  😊

또한

엔티티의 `createNotice`: 새로운 도메인 객체를 생성하는 비즈니스 로직
DTO의 `from`: 단순 데이터 변환

이라고 생각되어 굳이 바꾸지 않으셔도 될 것 같습니다 !!

Reply:
넵 감사합니다 다른사항 수정했어서 확인해주시고 승인 부탁드려요